packages/plugin-commands-patching/src/patch.ts

```diff
@@ -41,13 +69,16 @@ export async function handler (opts: PatchCommandOptions, params: string[]) {
     lockfileDir: opts.dir,
     preferredVersions: {},
     projectDir: opts.dir,
-    registry: (dep.alias && pickRegistryForPackage(opts.registries, dep.alias)) ?? opts.registries.default,
+    registry:
+      (dep.alias && pickRegistryForPackage(opts.registries, dep.alias)) ??
+      opts.registries.default,
   })
   const filesResponse = await pkgResponse.files!()
-  const tempDir = tempy.directory()
-  const userChangesDir = path.join(tempDir, 'user')
+
+  const patchPackageDir = createPackageDirectory(opts.path)
+  const userChangesDir = path.join(patchPackageDir, 'user')
   await Promise.all([
-    store.ctrl.importPackage(path.join(tempDir, 'source'), {
+    store.ctrl.importPackage(path.join(patchPackageDir, 'source'), {
```

Comment:
maybe the source dir should remain in the temp directory and only the user dir should be placed to the directory specified via the option.